src/main.c

```diff
@@ -194,6 +224,12 @@ void indirect_rv_halt()
 }
 #endif
 
+#if RV32_HAS(SYSTEM) && !defined(USE_ELF)
+/* forcely undefine MEM_SIZE to prevent any define in Makefile */
+#undef MEM_SIZE
+#define MEM_SIZE 512 * 1024 * 1024
+#endif
```

Comment:
`ENABLE_SDL` is enabled by default unless libsdl is not installed on the system. `ENABLE_FULL4G` will be enabled if `ENABLE_SDL` is active. However, since kernel emulation uses a dedicated memory layout, I explicitly undefine `MEM_SIZE` when kernel emulation is detected. Does this a good way, or would it be better to handle this in the Makefile instead?

Reply:
Are `ENABLE_SDL`  and `ENABLE_SYSTEM` options mutually exclusive? (i.e., libsdl is used to support the userland graphic output) If so, we can disable libsdl in Makefile if it's in system simulation.

Reply:
> Are `ENABLE_SDL` and `ENABLE_SYSTEM` options mutually exclusive? (i.e., libsdl is used to support the userland graphic output) If so, we can disable libsdl in Makefile if it's in system simulation.

![image](https://github.com/user-attachments/assets/053f4eb6-9c4b-4434-8f1e-d2939fa031b1)

IMO, they are not mutually exclusive. I conducted a simple test by using trap-and-emulate for the SDL instruction from the guestOS, which just opens a white screen, and it works (see the image).

The issue here is that we need a flexible way to specify the dedicated `MEM_SIZE` when `ENABLE_SDL` is enabled, especially when the full 4GB is not required.

Reply:
AFAIK, the guestOS SDL trap-and-emulate is not currently supported. It should and will be in another PR.

Reply:
The intuitive approach is that even if `ENABLE_SDL` is enabled, `ENABLE_FULL4G` should remain disabled. The MEM_SIZE should depend on the specified use case.

Reply:
Awesome! I like the idea of executing SDL-oriented systems inside the emulated Linux guest.

Recent experience has shown that implementing a fully functional Virtio-GPU for semu is challenging. However, rendering in an SDL window would be more feasible, requiring less effort and allowing for a simpler implementation.